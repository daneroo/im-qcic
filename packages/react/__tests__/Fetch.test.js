import React from 'react'
import renderer from 'react-test-renderer'

import { Fetch } from '../src/index.js'

describe('Fetch', () => {
  test('import Fetch', () => {
    expect(typeof Fetch).toBe('function')
  })
  test('Fetch renders', (done) => {
    const component = renderer.create(
      <Fetch
        url='https://time.qcic.n.imetrical.com/'
        render={({ loading, error, data }) => {
          if (loading) return <p>Loading...</p>
          if (error) return <p>{error.name}: {error.message}</p>
          try {
            const time = Date.parse(data.time)
            if (Number.isNaN(time)) {
              data = 'Unable to parse Date'
            }
            data = { time: '2020-06-06T17:19:47.338Z' }
          } catch (error) {
            data = error.message
          }

          return (<pre>{JSON.stringify(data)}</pre>)
        }}
      />
    )

    setTimeout(() => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
      done()
    }, 1000)
  })
})

// function delay (ms) {
//   return new Promise(resolve => setTimeout(resolve, ms))
// }
