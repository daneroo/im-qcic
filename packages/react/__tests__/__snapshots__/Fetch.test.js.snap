// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch Fetch renders 1`] = `
<pre>
  {"time":"2020-06-06T17:19:47.338Z"}
</pre>
`;
